test/timecop_test.rb

```diff
@@ -689,8 +689,105 @@ def test_thread_safe_timecop_returns_after_block
+    def test_process_clock_gettime_monotonic
+      Timecop.freeze do
+        assert_same monotonic, monotonic, "CLOCK_MONOTONIC is not frozen"
+      end
+
+      initial_time = monotonic
+      Timecop.freeze(-0.5) do
+        assert_operator(monotonic, :<, initial_time, "CLOCK_MONOTONIC is not traveling back in time")
+      end
+    end
+
+    def test_process_clock_gettime_monotonic_nested
+      Timecop.freeze do
+        parent = monotonic
+
+        sleep(0.01)
+
+        delta = 0.5
+        Timecop.freeze(delta) do
+          child = monotonic
+          assert_equal(child, parent + delta, "Nested freeze not working for monotonic time")
+        end
+      end
+    end
+
+    def test_process_clock_gettime_monotonic_travel
+      initial_time = monotonic
+      Timecop.travel do
+        refute_same monotonic, monotonic, "CLOCK_MONOTONIC is frozen"
+        assert_operator(monotonic, :>, initial_time, "CLOCK_MONOTONIC is not moving forward")
+      end
+
+      Timecop.travel(-0.5) do
+        refute_same monotonic, monotonic, "CLOCK_MONOTONIC is frozen"
+        assert_operator(monotonic, :<, initial_time, "CLOCK_MONOTONIC is not traveling properly")
+      end
+    end
+
+    def test_process_clock_gettime_monotonic_scale
+      scale = 4
+      sleep_length = 0.25
+      Timecop.scale(scale) do
+        initial_time = monotonic
+        sleep(sleep_length)
+        expected_time = initial_time + (scale * sleep_length)
+        assert_times_effectively_equal expected_time, monotonic, 0.1, "CLOCK_MONOTONIC is not scaling"
+      end
+    end
+
+    def test_process_clock_gettime_realtime
+      Timecop.freeze do
+        assert_same realtime, realtime, "CLOCK_REALTIME is not frozen"
+      end
+
+      initial_time = realtime
+      Timecop.freeze(-20) do
+        assert_operator(realtime, :<, initial_time, "CLOCK_REALTIME is not traveling back in time")
+      end
+    end
+
+    def test_process_clock_gettime_realtime_travel
+      initial_time = realtime
+      Timecop.travel do
+        refute_equal realtime, realtime, "CLOCK_REALTIME is frozen"
+        assert_operator(realtime, :>, initial_time, "CLOCK_REALTIME is not moving forward")
+      end
+
+      Timecop.travel(Time.now - 0.1) do
+        refute_equal realtime, realtime, "CLOCK_REALTIME is frozen"
+        assert_operator(realtime, :<, initial_time, "CLOCK_REALTIME is not traveling properly")
+        sleep 0.1
+        assert_operator(realtime, :>, initial_time, "CLOCK_REALTIME is not traveling properly")
+      end
+    end
+
+    def test_process_clock_gettime_realtime_scale
+      scale = 4
+      sleep_length = 0.25
+      Timecop.scale(scale) do
+        initial_time = realtime
+        sleep(sleep_length)
+        assert_operator(initial_time + scale * sleep_length, :<, realtime, "CLOCK_REALTIME is not scaling")
+      end
+    end
+  end
+
   private
 
+  if RUBY_VERSION >= '2.1.0'
+    def monotonic
+      Process.clock_gettime(Process::CLOCK_MONOTONIC)
```

Comment:
We're always using the default nanosecond unit for all our tests. I think it would be nice to have a test where we freeze and then make sure that Process.clock_gettime is converting the units correctly for seconds, milliseconds, etc.